### PR TITLE
{chem}[intel/2017b] Amber-16-Ambertools-17-patchlevel-8-12

### PR DESCRIPTION
**Pull request by akesandgren**
(created using `eb --new-pr`)
Amber 16.12 AmberTools 17.8 with a lot of patches to make the build as correct as possible, and make tests work

### CODE REVIEW

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 13 out of 13 (13 easyconfigs in this PR)
b-cn1522.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, Python 2.7.12
See https://gist.github.com/92253c7199a7b17e87e613ef356fc635 for a full test report.

File: easybuild/easyconfigs/i/impi/impi-2017.3.196-iccifortcuda-2017b.eb [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+name = 'impi'
+version = '2017.3.196'
+
+homepage = 'http://software.intel.com/en-us/intel-mpi-library/'
+description = """The Intel(R) MPI Library for Linux* OS is a multi-fabric message
+ passing library based on ANL MPICH2 and OSU MVAPICH2. The Intel MPI Library for
+ Linux OS implements the Message Passing Interface, version 2 (MPI-2) specification."""
+
+toolchain = {'name': 'iccifortcuda', 'version': '2017b'}
```
Comment (boegel) [inline easybuild/easyconfigs/i/impi/impi-2017.3.196-iccifortcuda-2017b.eb]: @akesandgren This easyconfig + the `iccifortcuda` one serve no purpose anymore?

File: easybuild/easyconfigs/c/CUDA/CUDA-9.1.85-iccifort-2017.4.196-GCC-6.4.0-2.28.eb [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+version = '9.1.85'
+nv_version = '387.26'
+
+homepage = 'https://developer.nvidia.com/cuda-toolkit'
+description = """CUDA (formerly Compute Unified Device Architecture) is a parallel
+ computing platform and programming model created by NVIDIA and implemented by the
+ graphics processing units (GPUs) that they produce. CUDA gives developers access
+ to the virtual instruction set and memory of the parallel computational elements in CUDA GPUs."""
+
+toolchain = {'name': 'iccifort', 'version': '2017.4.196-GCC-6.4.0-2.28'}
```
Comment (boegel) [inline easybuild/easyconfigs/c/CUDA/CUDA-9.1.85-iccifort-2017.4.196-GCC-6.4.0-2.28.eb]: @akesandgren This one is also no longer needed?

File: easybuild/easyconfigs/p/Python/Python-2.7.13-intel-2017b.eb [hunk 3]
```diff
@@ -0,0 +1,141 @@
+name = 'Python'
+version = '2.7.13'
+
+homepage = 'http://python.org/'
+description = """Python is a programming language that lets you work more quickly and integrate your systems
+ more effectively."""
+
+toolchain = {'name': 'intel', 'version': '2017b'}
```
Comment (boegel) [inline easybuild/easyconfigs/p/Python/Python-2.7.13-intel-2017b.eb]: @akesandgren This easyconfig is not actually needed in this context?

**Comment by akesandgren** [comment]
Currently fails with the recently merged intelcuda/2017b. Works with the intelcuda/2017b I initially had in this PR.
Will figure out why asap.

**Comment by akesandgren** [comment]
Doh, it doesn't like CUDA 9.0.176 (my intelcuda version happened to have 9.1)
So lets scrap intelcuda/2017b in this PR. And I'll make a new one with 2018b instead.

The important part here is the patches to Amber(Tools) not the specific builds.

**Comment by akesandgren** [comment]
Test report by @akesandgren
**SUCCESS**
Build succeeded for 5 out of 5 (5 easyconfigs in this PR)
b-cn1522.hpc2n.umu.se - Linux ubuntu 16.04, Intel(R) Xeon(R) Gold 6132 CPU @ 2.60GHz, Python 2.7.12
See https://gist.github.com/8ff48ff24ffa1097b338396a0ab17015 for a full test report.

**Comment by boegel** [comment]
I don't have access to the necessary sources to test this. Maybe @migueldiascosta does?

**Comment by migueldiascosta** [comment]
I manually fetched `Amber` and `AmberTools`, but I'm having some problem when `update_amber` tries to fetch the patches for `AmberTools`

@akesandgren it seems it's possible for `update_amber` to use offline patches, would it make sense for easybuild to fetch the patches instead of letting `amber_update` do it?

**Comment by migueldiascosta** [comment]
@akesandgren the temporary kind, I guess, now it's building...

**Comment by migueldiascosta** [comment]
Test report by @migueldiascosta
**SUCCESS**
Build succeeded for 1 out of 1 (1 easyconfigs in this PR)
grc-cluster1 - Linux centos 6.10, Intel(R) Xeon(R) CPU E5-2640 0 @ 2.50GHz, Python 2.7.14
See https://gist.github.com/f2a7e94ef03288e7dbc72db88c16f668 for a full test report.

**Comment by migueldiascosta** [comment]
@akesandgren why not Amber16.15 and AmberTools17.10, the latest patchlevel for those versions?

**Comment by migueldiascosta** [comment]
@akesandgren fair enough, just wondering if there was a(nother) reason (I don't know much about Amber)

fwiw, they don't seem to conflict with yours and I'm testing building with them, so far so good

**Comment by migueldiascosta** [comment]
this looks good to me - additionally, I was able to update the patchlevel and also add cuda support with no issues

currently running the benchmark suite mostly for my own benefit, will merge this today if nothing else comes up

**Comment by migueldiascosta** [comment]
Going in, thanks @akesandgren!

**Pull request merged**